### ISSUE TITLE
Implement popular optimizers

### ISSUE THREAD

**Issue by jessegrabowski** [jessegrabowski/pytensor-ml#7]
We should have most or all of the widely used optimizers in the ML literature:

- [ ] Adadelta
- [ ] Adagrad
- [ ] Adam
- [ ] AdamW 
- [ ] SparseAdam
- [ ] Adamax
- [ ] ASGD
- [ ] LBFGS
- [ ] NAdam
- [ ] RAdam
- [ ] RMSprop
- [ ] Rprop
- [ ] SGD

Currently only Adam and SGD are implemented, and using design decision that we can revisit. Most of these are implemented in PyMC, taken from the Lasagne library. We could just reuse these instead of rewriting our own. The most important thing is that the optimizers are easily "extendable" to include things like gradient clipping #5 and learning rate scheduling #6

**Comment by HangenYuu** [comment]
I am not sure about little circle, but I see that people can track issues with PRs, but that requires someone with write access to edit the comment by adding #title_number to the task item after opening the PR.
![image](https://github.com/jessegrabowski/pytensor-ml/assets/97870091/3a9b8367-a5c6-416f-9dca-7112fa82c5b6)

**Comment by jessegrabowski** [comment]
Like that?

**Comment by HangenYuu** [comment]
I think I need to change my PR to just "AdamW" for it to work like that. So for the field you will just write `- [ ] #8`. But any case, save that for later since I will close the PR #8 just now to split it properly into 3 individual PRs. It will take some time since I have to revert the Adadelta change that I committed 😅.
